### PR TITLE
Extend nic with mac

### PR DESCRIPTION
**Pull request by Vatson112**
## Please describe the change you are making

Allow manage and read MAC address of VM NIC interfaces.

Issue reference in terraform client: [issue](https://github.com/oVirt/terraform-provider-ovirt/issues/484) 
...

## Are you the owner of the code you are sending in, or do you have permission of the owner?

Yes
...

## The code will be published under the BSD 3 clause license. Have you read and understood this license?

Yes
...

### CODE REVIEW

**Comment by engelmi** [comment]
> @engelmi can you run live tests on this PR please? The code looks good to me.

I am on, will take a while, though. 

@Vatson112 the linting is currently failing due to [gci](https://github.com/daixiang0/gci). This results from using the latest golangci-lint version - and recently v1.50.1 was released. Formatting with `goimports` should solve this issue. Could you apply it to the affected files?

**Comment by Vatson112** [comment]
Hi @engelmi!

I cannot solve the problem with `goimports` or `gci`. And dont understand the problem with `golangci-lint`.

Running `golangci-lint` locally I see the problem with 3 files. Take for example 1st.
```bash
> golangci-lint run -c .golangci.yml
util_wait_for_job_finished.go:16: File is not `gci`-ed with --skip-generated -s standard,default (gci)
//     correlationID := fmt.Sprintf("image_transfer_%s", utilrand.String(5))
//     conn.
//         SystemService().
//         DisksService().
//         DiskService(diskId).
//         Update().
//         Query("correlation_id", correlationID).
//         Send()
new.go:80: File is not `gci`-ed with --skip-generated -s standard,default (gci)
//   url
new.go:84: File is not `gci`-ed with --skip-generated -s standard,default (gci)
//   username
```
I tried `gci diff` to find problem. But it doesnt show anything.
```bash
> ~/go/bin/gci  diff  --skip-generated  util_wait_for_job_finished.go 
```
Also I tried `goimports`, and It looks the same.
```bash
> ~/go/bin/goimports -d  util_wait_for_job_finished.go
``` 
So, I need your help to understand how to fix this.

**Comment by engelmi** [comment]
Hi @Vatson112, 
I confused the tool. It is not `goimports`, but `gofmt` - a simple 
```bash
gofmt  -w newmock.go new.go util_wait_for_job_finished.go
```
should suffice. (I think these are the files the linter is failing)

**Comment by Vatson112** [comment]
Hi! 

Nope, `gofmt` didn't change files with your command. Also neither show diff, nor list files whose formatting differs from gofmt, show anything.
```bash
> gofmt -d util_wait_for_job_finished.go
> gofmt -d newmock.go
> gofmt -d new.go 
```

```bash
> gofmt -l .
```

**Comment by engelmi** [comment]
Interesting, locally this did fix it for me - its just about formatting the doc comments. 
Which Go version do you use? My guess would be that this is due to [gofmt and doc changes in 1.19](https://go.dev/doc/go1.19#tools) as [golangci-lint uses 1.19](https://github.com/golangci/golangci-lint/blob/master/go.mod#L3).

**Comment by Vatson112** [comment]
I use `go1.19.2` and `golangci-lint  version 1.50.1`.

I think I fixed the issue with `golangci-lint`.

Firstly I tried to use docker container with `go1.16`, but seems it is not a problem with go version.

Then,  I found `golangci-lint` `--fix` option, so I tried to use it firstly with `go1.16` and it was fixed all linter problem, then I tried the same with `go1.19` and It also works great.
So my full command is.
```bash
golangci-lint  run -c .golangci.yml  --whole-files  --fix
```

I'll write commit now. And we will check CI.

File: nic_test.go [hunk 1]
```diff
@@ -31,6 +32,11 @@ func assertCanCreateNIC(
 	if nic.VMID() != vm.ID() {
 		t.Fatalf("VM ID mismatch between NIC and VM (%s != %s)", nic.VMID(), vm.ID())
 	}
+	if params != nil {
+		if nic.Mac() != params.Mac() {
+			t.Fatalf("Failed to create NIC with custom mac address: %s", nic.Mac())
```
Comment (engelmi) [inline nic_test.go:37]:
```suggestion
		if params.Mac() != "" && params.Mac() != nic.Mac() {
			t.Fatalf("Failed to create NIC with custom mac address. Expected '%s', but created mac is '%s'", params.Mac(), nic.Mac())
```
The additional empty string check on the mac param is required here. If the mac param is not set, a mac for the nic will be generated anyway when testing against a real oVirt engine - and cause this assert func to fail (for example in [TestVMNICCreation](https://github.com/oVirt/go-ovirt-client/blob/main/nic_create_test.go#L21-L26)

**Review by engelmi — CHANGES_REQUESTED** [review]
Finally got the time (and resources) to test this against a real oVirt engine. Found only one minor issue. As soon as this is fixed, we can finally merge this PR :)

**Review by engelmi — APPROVED** [review]

**Pull request merged**